packages/zowe-explorer-api/src/vscode/ZoweVsCodeExtension.ts

```diff
@@ -106,6 +150,18 @@ export class ZoweVsCodeExtension {
         return await vscode.window.showInputBox(inputBoxOptions);
     }
 
+    private static async saveCredentials(profile: imperative.IProfileLoaded): Promise<boolean> {
+        let saved = false;
+        const saveButton = "Save Credentials";
+        const message = `Save entered credentials in plain text for future use with profile ${profile.name}?\nSaving credentials will update the local information file.`;
+        await vscode.window.showInformationMessage(message, { modal: true }, ...[saveButton]).then((selection) => {
+            if (selection) {
+                saved = true;
+            }
+        });
+        return saved;
```

Comment:
It's best to use immutable variables or return once you know the final value instead of declaring a temporary variable and mutating it through the function. Mutating the variable throughout a function creates code that's more complicated and easier to accidentally introduce a bug. 
In this example it would be something like:
```
    private static async saveCredentials(profile: imperative.IProfileLoaded): Promise<boolean> {
        const saveButton = "Save Credentials";
        const message = `Save entered credentials in plain text for future use with profile ${profile.name}?\nSaving credentials will update the local information file.`;
        await vscode.window.showInformationMessage(message, { modal: true }, ...[saveButton]).then((selection) => {
            if (selection) {
                return true
            }
        });
        return false;
```
Here's one reference article: https://zellwk.com/blog/dont-reassign/

Reply:
addressed in commit [df19d4c](https://github.com/zowe/vscode-extension-for-zowe/pull/1873/commits/df19d4c40f86f542b496362f8e7ed0037a2751e7)